DOCSP-43838 - Upgrade Guides (#239)
mongodb/docs-csharp: source/upgrade/v3.txt

.. _csharp-upgrade-v3:

======================
Upgrade to Version 3.0
======================

.. facet::
   :name: genre
   :values: reference

.. meta::
   :keywords: update, breaking change, releases

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 1
   :class: singlecol

Overview
--------

This page describes the changes you might need to make to your
application when you upgrade the {+driver-short+} to version 3.0.

How to Upgrade
--------------

> Each section of this page lists the potential breaking changes introduced by a driver version.

Reviewer: S:
```suggestion
This page lists the potential breaking changes introduced by the {+driver-short+} version 3.0.
```

To upgrade the {+driver-short+} to version 3.0, follow these steps:

1. Review the :ref:`Compatibility <csharp-compatibility-tables>` page to ensure the new
   driver version is compatible with the {+mdb-server+} versions your application connects
   to and the .NET or {+framework+} version your application runs on.

#. If you're using a 2.x version of the {+driver-short+}, upgrade to
   v{+last-version-2-number+}. To do so, follow the
   :ref:`v2.x upgrade guide. <csharp-upgrade-v2>`

#. Address the breaking changes described in the

>    :ref:`Breaking Changes by Version <csharp-breaking-changes-v3>` section.

Reviewer: Since the header name changed
```suggestion
   :ref:`<csharp-breaking-changes-v3>` section.
```

.. _csharp-upgrade-v3:

======================
Upgrade to Version 3.0
======================

.. facet::
   :name: genre
   :values: reference

.. meta::
   :keywords: update, breaking change, releases

.. contents:: On this page
   :local:
   :backlinks: none
   :depth: 2
   :class: singlecol

Overview
--------

This page describes the changes you might need to make to your
application when you upgrade the {+driver-short+} to version 3.0.

How to Upgrade
--------------

Each section of this page lists the potential breaking changes introduced by a driver version.
To upgrade the {+driver-short+} to version 3.0, follow these steps:

1. Review the :ref:`Compatibility <csharp-compatibility-tables>` page to ensure the new
   driver version is compatible with the {+mdb-server+} versions your application connects
   to and the .NET or {+framework+} version your application runs on.

#. If you're using a 2.x version of the {+driver-short+}, upgrade to
   v{+last-version-2-number+}. To do so, follow the

>    :ref:`v2.x upgrade guide <csharp-upgrade-v2>` guide.

Reviewer: ```suggestion
   :ref:`v2.x upgrade guide <csharp-upgrade-v2>`.
```

Author: good catch

#. Address any breaking changes between the driver version your
   application uses and your target upgrade version in the

>    :ref:`Breaking Changes by Version <csharp-breaking-changes-v3>` section.

Reviewer: Q: Since they have to upgrade to v2.9 first, wouldn't this step just be to address the changes listed below? Judging by the Slack thread w/ Rishit, it seemed to me like we will only be showing 3.0 and no minor versions on this page?

Author: good point. i didn't think to change that after rishit suggested updating the page title

   .. example::
    
      If you are upgrading the driver from v2.14 to v3.0, first use the v2.x upgrade guide
      to upgrade your driver to v{+last-version-2-number+}. Then, address all breaking
      changes for v3.0.

.. _csharp-breaking-changes-v3:

Breaking Changes by Version
---------------------------

The following sections describe the potential breaking changes introduced by each driver

> version. If a driver version is not included in this section, it doesn't introduce any

Reviewer: Similar Q/S here. If this page will only have 3.0, we can simplify this to state that rather than saying "introduced by each driver version"
